staging/src/k8s.io/apimachinery/pkg/api/meta/conditions.go

```diff
@@ -28,7 +28,7 @@ import (
 //    newCondition, LastTransitionTime is set to now if the new status differs from the old status)
 // 2. if a condition of the specified type does not exist (LastTransitionTime is set to now() if unset, and newCondition is appended)
 func SetStatusCondition(conditions *[]metav1.Condition, newCondition metav1.Condition) {
-	if conditions == nil {
```

Comment:
why can it never be nil?

Reply:
Since "conditions" argument is a pointer to a slice of Condition, it has (Type,Val) = (*[]metav1.Condition, nil) structure. Therefore the comparison of type pointer to a slice of Condition to nil will not succeed.

GopherCon 2016: Francesc Campoy - Understanding nil can help you understanding.
https://youtu.be/ynoY2xz-F8s?t=782

Correct me if I'm wrong!

Reply:
var a *[]metav1.Condition
SetStatusCondition(a, ...)

then conditions == nil, right?

Reply:
That is right. I missed that one could use that way.

For building a new controller, I was following the example in [Condition comment ](https://github.com/kubernetes/apimachinery/blob/v0.19.2/pkg/apis/meta/v1/types.go#L1363).

I added the Conditions in Status as follows.
```
type VirtualMachineStatus struct {
	// Conditions represent the latest available observations of an VirtualMachine object's state
	//+patchMergeKey=type
	//+patchStrategy=merge
	//+listType=map
	//+listMapKey=type
	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
}
```

Therefore I used the address of Conditions for SetStatusCondition.
```
meta.SetStatusCondition(&vm.Status.Conditions, v1.Condition{
	Type:    "Ready",
	Status:  v1.ConditionUnknown,
	Reason:  "Init",
	Message: "Initializing",
})
```

In this case, the nil comparison will always be false. But, I was not thinking of your point.
Should I close this pull request?

Reply:
sure, feel free to close the pr.

Reply:
Thanks!